.github/workflows/ci_tests.yml

```diff
@@ -0,0 +1,218 @@
+      run: |
+        echo "Is PR a draft?"
+        echo ${{ toJSON(github.event.pull_request.draft) }}
+        if [[ ${{ toJSON(github.event.pull_request.draft) }} == true ]]; then "false" ; else "true"; fi
+
+  # PR must be assigned to be merged.
+  # This job will fail if this is not the case.
+  check_pull_request_is_assigned:
+    needs: [check_pull_request_is_not_a_draft]
+    runs-on: ubuntu-22.04
+    steps:
+    # Assignee ID will be null if PR is not assigned
+    - name: Check that the PR is assigned
+      run: |
+        id=$(curl -H "Accept: application/vnd.github+json" \
+           https://api.github.com/repos/${{ github.repository }}/pulls/${{ github.event.number }} \
+           | jq '.assignee.id')
+        echo "Is PR Assigned? (Check for Assignee ID)"
+        echo $id
+        if [[ $id != null ]]; then "true" ; else "false"; fi
+
+  check_submodules:
+    needs: [check_pull_request_is_not_a_draft]
+    runs-on: ubuntu-22.04
+    steps:
+    # The integrated test submodule repository contains large data (using git lfs).
+    # To save time (and money) we do not let Github Actions automatically clone all our (lfs) subrepositories and do it by hand.
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        # Let script update submodules; Github Actions submodule history causes error
+        submodules: false
+        lfs: false
+    - name: Check that submodules are up to date
+      run: "scripts/test_submodule_updated.sh"
+
+  code_style:
+    needs: [check_pull_request_is_not_a_draft]
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+        lfs: false
+    - name: Check style
+      env:
+        DOCKER_REPOSITORY: geosx/ubuntu20.04-gcc9
+        CMAKE_BUILD_TYPE: Release
+        BUILD_AND_TEST_ARGS: --test-code-style
+      run: ./scripts/ci_build_and_test.sh
+
+  documentation:
+    needs: [check_pull_request_is_not_a_draft]
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+      with:
+        submodules: true
+        lfs: false
+    - name: Check documentation
+      env:
+        DOCKER_REPOSITORY: geosx/ubuntu20.04-gcc9
+        CMAKE_BUILD_TYPE: Release
+        BUILD_AND_TEST_ARGS: --test-documentation
+      run: ./scripts/ci_build_and_test.sh
+
+
+  linux_builds:
+    name: ${{ matrix.name }}
+    runs-on: ubuntu-22.04
+    needs: [check_pull_request_is_not_a_draft]
+    strategy:
+
+      # In-progress jobs will not be cancelled if there is a failure
+      fail-fast : false
+      matrix:
+        include:
+          - name: Ubuntu CUDA debug (20.04, clang 10.0.0 + gcc 9.4.0, open-mpi 4.0.3, cuda-11.2.152)
+            DOCKER_REPOSITORY: geosx/ubuntu20.04-clang10.0.0-cuda11.2.152
+            CMAKE_BUILD_TYPE: Debug
+            BUILD_AND_TEST_ARGS: "--disable-unit-tests --build-exe-only --disable-schema-deployment"
```

Comment:
if we don't have any time limit anymore then, I think we should not need `--build-exe-only` and we could also remove `--disable-unit-tests` I think.